src/storage/IpfsStorage.ts

```diff
@@ -63,8 +65,23 @@ export class IpfsStorage implements IStorage {
     return `ipfs://${cid}/`;
   }
 
+  public async uploadBatchWithFileNames(
+    files: FileOrBufferWithNames[],
+    contractAddress?: string,
+  ): Promise<string> {
+    const cid = await this.uploadBatchWithCid(true, files, contractAddress);
+
+    return `ipfs://${cid}/`;
+  }
+
   private async uploadBatchWithCid(
-    files: Buffer[] | string[] | FileOrBuffer[] | File[],
+    withFileNames = false,
```

Comment:
Please put this argument at the end of the function parameter list so that it doesn't disrupt existing function calls.

This is a common programmer mistake when trying to retain backwards compatibility. Here's an example.

Say there's a function:

```ts
function add(x: number, y: number, optionA = false)
```

Lets say a developer *John* used this code and called this function like this:

```ts
add(1,2, true)
```

If we add a parameter to the front of the function like this, calls to the function will still work **but every single instance where the function was called needs to be updated because the ordering has changed**.

```js
function add(x: number, y: number, newOption = false, optionA = false)
```

Lets look back at that Johns code:

```js
add(1,2, true)
```

Now John doesn't realize it, but he's accidentally passing `true` for `newOption` because it took the position of `optionA` in the arg list.

---

The backwards compatible change would be to **preserve the ordering of existing parameters** like this:


```js
function add(x: number, y: number, optionA = false, newOption = false)
```

Reply:
Gotcha thanks! But I thought about this and it is a private function and we need `fileStartNumber` when `withFileNames` is true. so `withFileNames` has to be before `fileStartNumber` right?